weave/panels_py/panel_openai_monitor.py

```diff
@@ -108,9 +108,64 @@
 #         # table.add_column(lambda row: row["messages"][-1]["content"], "Message")
 #         return table
 
+API_COST_INPUT_TYPE = types.TypedDict(
+    {
+        "output": types.optional(
+            types.TypedDict(
+                {
+                    "model": types.optional(types.String()),
+                }
+            )
+        ),
+        "summary": types.optional(
+            types.TypedDict(
+                {
+                    "prompt_tokens": types.optional(types.Number()),
+                    "completion_tokens": types.optional(types.Number()),
+                }
+            )
+        ),
+    }
+)
 
-def cost(row: dispatch.RuntimeOutputNode) -> dispatch.RuntimeOutputNode:
-    return row["summary.total_tokens"] * 0.0015e-3
+
+@weave.op(weavify=True, input_type={"record": API_COST_INPUT_TYPE})  # type: ignore
+def openai_request_cost(record) -> float:  # type: ignore
+    model = record["output.model"]
+    pt = record["summary.prompt_tokens"]
+    ct = record["summary.completion_tokens"]
+    cost_per_1000 = weave.ops.case(
```

Comment:
this is really cool that we can mix op calls directly in for non-weavifiable parts of logic